tests/ctl/service_account_test.py

```diff
@@ -1,127 +1,151 @@
+from typing import TYPE_CHECKING
+
+from grouper.constants import USER_ADMIN
 from grouper.group_service_account import get_service_accounts
 from grouper.models.group import Group
 from grouper.models.service_account import ServiceAccount
-from tests.ctl_util import call_main
-from tests.fixtures import (  # noqa: F401
-    graph,
-    groups,
-    service_accounts,
-    session,
-    standard_graph,
-    users,
-)
-
-
-def test_service_account_create(groups, service_accounts, session, tmpdir, users):  # noqa: F811
-    machine_set = "foo +bar -(org)"
-    description = "this is a service account.\n\n it is for testing"
-    security_team_group = Group.get(session, name="security-team")
-    good_actor_username = "gary@a.co"
-    good_service_account_name = "good-service@a.co"
-
-    assert ServiceAccount.get(session, name=good_service_account_name) is None
-    assert get_service_accounts(session, security_team_group) == []
-    # no-op if non-existing actor
-    call_main(
-        session,
-        tmpdir,
+from tests.ctl_util import run_ctl
+
+if TYPE_CHECKING:
+    from tests.setup import SetupTest
+
+
+def test_create(setup):
+    # type: (SetupTest) -> None
+    with setup.transaction():
+        setup.add_user_to_group("gary@a.co", "some-group")
+        setup.add_user_to_group("gary@a.co", "other-group")
+
+    run_ctl(
+        setup,
         "service_account",
         "--actor",
-        "no-such-actor@a.co",
+        "gary@a.co",
         "create",
-        good_service_account_name,
-        security_team_group.groupname,
-        machine_set,
-        description,
+        "good-service@a.co",
+        "some-group",
+        "foo +bar -(org)",
+        "this is a service account.\n\n it is for testing",
     )
-    # ... or if bad account name
-    call_main(
-        session,
-        tmpdir,
+    service_account = ServiceAccount.get(setup.session, name="good-service@a.co")
+    assert service_account is not None
+    assert service_account.user.name == "good-service@a.co"
+    assert service_account.machine_set == "foo +bar -(org)"
+    assert service_account.description == "this is a service account.\n\n it is for testing"
+    group = Group.get(setup.session, name="some-group")
+    assert get_service_accounts(setup.session, group) == [service_account]
+
+    # If the account already exists, creating it again does nothing.
+    #
+    # TODO(rra): Change this behavior to return an error message instead, since it's confusing.
```

Comment:
Also general question rather than review:

Do you have a strategy for actually go backing and hitting TODOs? I avoid leaving them in final checked-in code because I'm pretty sure I'll just forget about them, but they do seem very useful in certain cases

Reply:
In general, I agree with you most of the time. In this specific case, subsequent diffs in this same patch series are going to fix the TODOs, so they won't be there long.

I will sometimes leave long-lived TODOs in the code not because I have any specific plans to go back and fix them but because I want to make explicit to someone reading the code that this bit of code is an error, or at least sub-optimal, and just hasn't been fixed. Otherwise, it can be hard to be sure whether the code is actually ideal because of some subtlety that isn't obvious. One can explain that in English, but TODO makes it totally clear that no, this is something that should be fixed, but there just hasn't been time yet.